tn_api_v2_core.yml

```diff
@@ -0,0 +1,947 @@
+            properties:
+                passenger_type_code:
+                    type: string
+                    description: Passenger type code. Example = "ADT"
+                base_price:
+                    type: number
+                total_price:
+                    type: number
+                taxes:
+                    type: number
+                tax_breakdown:
+                    type: array
+                    items:
+                        $ref: '#/components/schemas/DatasourceTaxBreakdownDetails'
+
+        DatasourceTaxBreakdownDetails:
+            type: object
+            description: Breakdown of taxes amount into their categories
+            properties:
+                category:
+                    type: string
+                    description: Tax category. Example = "CA"
+                amount:
+                    type: number
+                    description: Tax amount. Example = 5.295
+
+        GenerateSolutionsResponse:
+            title: Generate Solutions Response
+            type: object
+            properties:
+                trip_id:
+                    type: string
+                    description: A unique key used for identification and when booking a trip.
+                    example: "fb3d8eff432a329dc21fad8449c686fa93e5ce24"
+                itineraries:
+                    $ref: '#/components/schemas/DatasourceItineraryInfo'
+
+        DatasourceItineraryInfo:
+            type: array
+            description: Array of itineraries sorted by weight.
+            items:
+                properties:
+                    weight:
+                        type: float
+                        description: Weight of the segment (time-value considered).
+                        example: 239
+                    total_price:
+                        type: float
+                        description: Price of the itinerary (fare and tax included).
+                        example: 197.1
+                    transportation_time:
+                        type: integer
+                        description: Total transportation time for the itinerary
+                        example: 425
+                    structure:
+                        type: Array of ints and/or Array of tuple/or Array of tuple and int
+                        description: |
+                            "Array representing the structure of the trip in terms of one-ways and open-jaw segments. 
+                            For example: [(0, 1), (3, 4), 2] means there would be three segments. One openjaw with key “01”, 
+                            one with key “34” and a one-way with key ‘2”."
+                            See Price Confirm Report segment_ids field for more detail.
+                        example: "[(0, 1), 2]"
+                    segment_metadata:
+                        type: array
+                        description: Contains extra information about the segment that was sent by the OTA
```

Comment:
I thought we were making this a dictionary, not an array. Also, if it's optional lets say in the description they can put anything they want in there (and some examples) and we'll return it but we shouldn't spell out all the datastructure because we actually don't enforce that data structure.

Reply:
Should also add segment_metadata to the request

Reply:
the reason why i made this an array inside the response is because there could be multiple segments in the itinerary, so we will need to add those segment_metadata together to generate an itinerary.

Will add more description saying to the OTA that this will be unchanged for what they will be putting in!